src/main/java/beam/agentsim/events/handling/BeamEventsLogger.java

```diff
@@ -47,16 +48,46 @@ void iterationEnds() {
 
     protected void createEventsWriters() {
         int iterationNumber = matsimServices.getIterationNumber();
-        final int writeEventsInterval = beamServices.beamConfig().beam().outputs().writeEventsInterval();
-        final boolean writeThisIteration = (writeEventsInterval > 0) && (iterationNumber % writeEventsInterval == 0);
-        if (writeThisIteration) {
+
+        final BeamConfig.Beam.Outputs outputsConfig = beamServices.beamConfig().beam().outputs();
+
+        final int writeEventsInterval = outputsConfig.writeEventsInterval();
+        final boolean writeAllEvents = (writeEventsInterval > 0) && (iterationNumber % writeEventsInterval == 0);
+
+        final String extraEventsToWrite = outputsConfig.events().extraEventsToWrite().getOrElse(() -> "");
+        final List<Class<?>> extraEventsClasses = getEventsClasses(extraEventsToWrite);
+        final boolean writeExtraEvents = !extraEventsClasses.isEmpty();
+
+        if (writeAllEvents || writeExtraEvents) {
```

Comment:
Should we allow writeAll and writeExtra? Or at least check for overlap to avoid duplication of storage (maybe warn at minimum)? As to the underlying issue - I don't know that the naming elicits what it is doing. I prefer verbosity if needed, so maybe eventClassesToBeWrittenToSeparateOutputFiles?